### ISSUE TITLE
feat(ml): add log_loss, brier_score_loss, average_precision to compute_metrics_by_name

### ISSUE THREAD

**Issue by esperie** [terrene-foundation/kailash-py#342]
## Summary

`_shared.compute_metrics_by_name` supports 9 metrics (accuracy, f1, precision, recall, auc, mse, rmse, mae, r2) but is missing three commonly-used metrics that ASCENT M3 exercises need.

## Missing Metrics

| Metric | sklearn Function | ASCENT Usage |
|--------|-----------------|-------------|
| `log_loss` | `sklearn.metrics.log_loss` | 4 solutions (ex_2, ex_5, ex_6, ex_8) |
| `brier_score_loss` | `sklearn.metrics.brier_score_loss` | 4 solutions (ex_2, ex_3, ex_6, ex_8) |
| `average_precision` | `sklearn.metrics.average_precision_score` | 6 solutions (ex_2, ex_3, ex_5, ex_6, ex_7, ex_8) |

## Proposed Change

Add three entries to the `_METRIC_MAP` dict in `kailash_ml/engines/_shared.py`:

```python
"log_loss": lambda y_true, y_pred: sklearn.metrics.log_loss(y_true, y_pred),
"brier_score": lambda y_true, y_pred: sklearn.metrics.brier_score_loss(y_true, y_pred),
"average_precision": lambda y_true, y_pred: sklearn.metrics.average_precision_score(y_true, y_pred),
```

This allows `EvalSpec(metrics=["accuracy", "log_loss", "brier_score", "average_precision"])` to work.

## Priority

**P0** — Tiny change (~10 lines), large impact (4-6 exercise files can drop raw sklearn.metrics imports).

**Comment by esperie** [comment]
Resolved in 090c0e97 — feat(platform): full clearance
